src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -323,6 +365,101 @@ export class QuickBar extends LitElement {
     );
   }
 
+  private _generateNavigationCommands(): QuickBarItem[] {
+    const panelItems = this._generateNavigationPanelCommands();
+    const sectionItems = this._generateNavigationConfigSectionCommands();
+
+    return this._withNavigationActions([...panelItems, ...sectionItems]);
+  }
+
+  private _generateNavigationPanelCommands(): Omit<
+    QuickBarNavigationItem,
+    "action"
+  >[] {
+    return Object.keys(this.hass.panels).map((panelKey) => {
+      const panel = this.hass.panels[panelKey];
+
+      return {
+        text: this._getPanelText(panel),
+        icon: "hass:robot",
+        path: `/${panel.url_path}`,
+      };
+    });
+  }
+
+  private _generateNavigationConfigSectionCommands(): Partial<
+    QuickBarNavigationItem
+  >[] {
+    const items: NavigationInfo[] = [];
+
+    for (const sectionKey of Object.keys(configSections)) {
+      for (const page of configSections[sectionKey]) {
+        if (canShowPage(this.hass, page)) {
+          if (page.component) {
+            const info = this._getNavigationInfoFromConfig(
+              sectionKey,
+              page.component
+            );
+
+            if (info) {
+              items.push(info);
+            }
+          }
+        }
+      }
+    }
+
+    return items;
+  }
+
+  private _getNavigationInfoFromConfig(
+    sectionKey: string,
+    component: string
+  ): NavigationInfo | undefined {
+    const panel = configSections[sectionKey].find(
+      (section) => section.component === component
+    );
+
+    const shortCaption = this.hass.localize(
+      `ui.dialogs.quick-bar.commands.navigation.${component}`
+    );
+
+    if (panel && panel.translationKey && shortCaption) {
+      const caption = this.hass.localize(
+        "ui.dialogs.quick-bar.commands.navigation.navigate_to",
+        "panel",
+        shortCaption
+      );
+
+      return { ...panel, text: caption };
+    }
+
+    return undefined;
+  }
+
+  private _getPanelText(panel: Panels["panel"]) {
+    let translationKey = "";
+
+    if (!panel.title) {
+      switch (panel.component_name) {
+        case "profile":
+          translationKey = "panel.profile";
+          break;
+        case "lovelace":
+          translationKey = "panel.states";
+          break;
+      }
+    } else {
+      translationKey = `panel.${panel.title}`;
+    }
+
+    return this.hass.localize(
+      "ui.dialogs.quick-bar.commands.navigation.navigate_to",
+      "panel",
+      toTitleCase(this.hass.localize(translationKey))
+    );
+  }
```

Comment:
Can we move this helper to a generic place? Check <https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/data/panel.ts#L24>

Reply:
done